torchvision/prototype/datasets/_builtin/voc.py

```diff
@@ -0,0 +1,141 @@
+import functools
+import io
+import pathlib
+from typing import Any, Callable, Dict, List, Optional, Tuple
+from xml.etree import ElementTree
+
+import torch
+from torch.utils.data import IterDataPipe
+from torch.utils.data.datapipes.iter import (
+    Mapper,
+    TarArchiveReader,
+    Shuffler,
+    Filter,
+    Demultiplexer,
+)
+from torchdata.datapipes.iter import KeyZipper, LineReader
+from torchvision.datasets import VOCDetection
+from torchvision.prototype.datasets.utils import (
+    Dataset,
+    DatasetConfig,
+    DatasetInfo,
+    HttpResource,
+    OnlineResource,
+)
+from torchvision.prototype.datasets.utils._internal import (
+    path_accessor,
+    getitem,
+    INFINITE_BUFFER_SIZE,
+    path_comparator,
+)
+
+HERE = pathlib.Path(__file__).parent
+
+
+class VOC(Dataset):
+    @property
+    def info(self) -> DatasetInfo:
+        return DatasetInfo(
+            "voc",
+            homepage="http://host.robots.ox.ac.uk/pascal/VOC/",
+            valid_options=dict(
+                split=("train", "val", "test"),
+                year=("2012",),
+                task=("detection", "segmentation"),
+            ),
+        )
+
+    def resources(self, config: DatasetConfig) -> List[OnlineResource]:
+        if config.year == "2012":
+            if config.split == "train":
+                archive = HttpResource(
```

Comment:
I was looking to `HttpResource`. I am guessing I need to update `on_disk_cache` before you start to implement `HttpResource` to download and cache the Dataset in the `to_datapipe` method.

Reply:
Right now the download functionality is not enabled and we rely on the files already being present in the right directory.